src/DacpacTool/VersionChecker.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Threading;
+using System.Net.Http.Json;
+using NuGet.Versioning;
+using System.IO;
+using System.Globalization;
+using System.Net.Http;
+
+namespace MSBuild.Sdk.SqlProj.DacpacTool
+{
+
+    public class VersionChecker
+    {
+        private readonly IConsole _console;
+        private readonly IVersionProvider _versionProvider;
+
+        public VersionChecker(IConsole console, IVersionProvider versionProvider)
+        {
+            _versionProvider = versionProvider;
+            _console = console;
+        }
+
+        public async Task CheckForPackageUpdateAsync()
+        {
+            try
+            {
+                var timeout = TimeSpan.FromSeconds(2);
+
+                using var cts = new CancellationTokenSource(timeout);
+
+                var cacheFile = Path.Join(Path.GetTempPath(), "MSBuild.Sdk.SqlProj.tag-" + DateTime.Now.ToString("yyyy-MM-dd", CultureInfo.InvariantCulture) + ".txt");
```

Comment:
Now that I think of it, this is going to create a new file each day (you are using MSBuild.Sdk.SqlProj). It is probably relatively small, but we don't actively clean up these files anywhere, so over time this can accumulate a bit. Perhaps we just want a single file and use the last write time to determine if we need to refetch it?

Reply:
Good idea!